lang/src/test/kotlin/org/partiql/lang/eval/evaluatortestframework/ExprValueStrictEquals.kt

```diff
@@ -0,0 +1,146 @@
+package org.partiql.lang.eval.evaluatortestframework
+
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueType
+import org.partiql.lang.eval.bigDecimalValue
+import org.partiql.lang.eval.booleanValue
+import org.partiql.lang.eval.bytesValue
+import org.partiql.lang.eval.dateValue
+import org.partiql.lang.eval.intValue
+import org.partiql.lang.eval.name
+import org.partiql.lang.eval.numberValue
+import org.partiql.lang.eval.stringValue
+import org.partiql.lang.eval.timeValue
+import org.partiql.lang.eval.timestampValue
+
+/**
+ * Only used to verify the actual `ExprValue` is the same as the expected `ExprValue` in test
+ *
+ * Ensures 2 [ExprValue]s to have the same type & value
+ */
```

Comment:
I believe the doc comments could give a tad more insight into what you have figured out.  
What follows are some suggestions.  Fell free to modify, esp. make sure they don't say something that's actually wrong. 

/**
Strict equality that captures equality in the PartQL data model.
That is, two differently-constructed [ExprValue] values are never equal, unless they are bags or structs, where ordering of members is ignored. 
In particular, values are never equal if they belong to different types (such as numerics INT and FLOAT), making this equality different from [ExprValue.exprEquals].  

This implementation is currently only intended for use in tests, as it has not been validated for efficiency.   
*/